lib/resty/auto-ssl/storage_adapters/redis.lua

```diff
@@ -106,6 +106,30 @@ function _M.delete(self, key)
   return connection:del(prefixed_key(self, key))
 end
 
+function _M.keys_with_prefix(self, prefix)
+  local connection, connection_err = self:get_connection()
+  if connection_err then
+    return false, connection_err
+  end
+
+  local keys, err = connection:keys(prefixed_key(self, prefix .. "*"))
+
+  if keys and self.options["prefix"] then
+    local unprefixed_keys = {}
```

Comment:
If we change the `expiry` logic to fetch the date from the certificate using openssl, then I'm not sure we'll need to keep this function (although it could actually be useful for https://github.com/GUI/lua-resty-auto-ssl/pull/124#issuecomment-386942892). In any case, if we do keep this, then I'd just recommend shifting this duplicate logic to generate the `unprefixed_keys` outside of this method so it can more easily be shared between `keys_with_suffix` and `keys_with_prefix`.